core/src/main/java/hudson/model/FileParameterDefinition.java

```diff
@@ -99,14 +99,22 @@ private String getFileName(String possiblyPathName) {
         return possiblyPathName;
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     public ParameterValue createValue(CLICommand command, String value) throws IOException, InterruptedException {
         // capture the file to the server
-        FilePath src = new FilePath(command.checkChannel(),value);
         File local = File.createTempFile("jenkins","parameter");
-        src.copyTo(new FilePath(local));
+        String name;
+        if (value.isEmpty()) {
+            FileUtils.copyInputStreamToFile(command.stdin, local);
```

Comment:
I would also handle `null` and blank strings just in case. It will work if and only if there is one file parameter in the build, right? So for multiple params usage of old CLI is the only option?